model_compression_toolkit/core/common/quantization/quantization_params_generation/kl_selection.py

```diff
@@ -19,136 +19,8 @@
-        return _kl_error_function(y, range_min=range_min, range_max=threshold, n_bits=n_bits)
-
-    return {THRESHOLD: qparams_selection_tensor_search(_loss,
-                                                       tensor_data,
-                                                       n_bits,
-                                                       per_channel=per_channel,
-                                                       channel_axis=channel_axis,
-                                                       n_iter=n_iter,
-                                                       min_threshold=min_threshold)}
-
-
-def kl_selection_histogram(bins: np.ndarray,
-                           counts: np.ndarray,
-                           p: int,
-                           n_bits: int,
-                           min_value: float,
-                           max_value: float,
-                           constrained: bool = True,
-                           n_iter: int = 10,
-                           min_threshold: float = MIN_THRESHOLD,
-                           quant_error_method: qc.QuantizationErrorMethod = qc.QuantizationErrorMethod.KL) -> dict:
-    """
-    Compute the optimal threshold based on KL-divergence to quantize a histogram.
-    The error computation is based on KL-divergence to quantify the distributions
-    similarity of the histogram and its quantized histogram.
-
-    Args:
-        bins: Bins values of the histogram.
-        counts: Bins counts of the histogram.
-        p: p-norm to use for the Lp-norm distance (not used for this method).
-        n_bits: Number of bits to quantize the tensor.
-        min_value: Min value (not used for this method).
-        max_value: Max value (not used for this method).
-        constrained: Whether the threshold should be constrained or not.
-        n_iter: Number of iteration ot search for the threshold.
-        min_threshold: Minimal threshold to use if threshold is too small.
-        quant_error_method: an error function to optimize the parameters' selection accordingly (not used for this method).
-
-    Returns:
-        Optimal threshold to quantize the histogram based on KL-divergence.
-    """
-
-    def _loss(q_bins: np.ndarray,
-              q_count: np.ndarray,
-              _bins: np.ndarray,
-              _counts: np.ndarray,
-              threshold,
-              min_max_range  # dummy
-              ) -> np.float:
-        """
-        Compute the KL-divergence between two histograms.
-
-        Args:
-            q_bins: Bins values of the quantized histogram.
-            q_count: Bins counts of the quantized histogram.
-            _bins: Bins values of the original histogram.
-            _counts: Bins counts of the original histogram.
-            threshold: Threshold bins was quantized by.
-
-        Returns:
-            The KL-divergence between the two tensors histograms.
-        """
-
-        return _kl_error_histogram(q_bins,
```

Comment:
Why kl_error_histogram remain in this file and not moved to "error_histograms"? (not critical, I just wonder if it might be more correct if all other histogram error methods are there)

Reply:
We can. Just too many code in one file so I left it with separated file.

Reply:
I think it will be best to put all the "historgram" methods in the same place if we have a file for them anyway, don't you think?

Reply:
I agree with @ofirgo

Reply:
Done